unit_tests/test_aggregator.py

```diff
@@ -0,0 +1,415 @@
+        assert agg.accept_all_devices == accept
+
+    # *
+    @staticmethod
+    def test_constructor__device_uuid_list(aggregator_explicit):
+        assert aggregator_explicit.device_uuid_list == []
+
+    # *
+    @staticmethod
+    def test_constructor__aggregator_uuid(aggregator_explicit):
+        assert aggregator_explicit.aggregator_uuid is None
+
+    @staticmethod
+    def test_constructor__client_command_buffer():
+        with patch("gsy_e_sdk.aggregator.ClientCommandBuffer") as mocked_class:
+            Aggregator(aggregator_name=test_agg["name"])
+            mocked_class.assert_called_once()
+
+    @staticmethod
+    def test_constructor__connect_to_simulation():
+        with patch.object(Aggregator, "_connect_to_simulation") as mocked_method:
+            Aggregator(aggregator_name=test_agg["name"])
+            mocked_method.assert_called_once()
+
+    # *
+    @staticmethod
+    def test_constructor__latest_grid_tree(aggregator_explicit):
+        assert aggregator_explicit.latest_grid_tree == {}
+
+    # *
+    @staticmethod
+    def test_constructor__latest_grid_tree_flat(aggregator_explicit):
+        assert aggregator_explicit.latest_grid_tree_flat == {}
+
+    # *
+    @staticmethod
+    def test_constructor__area_name_uuid_mapping(aggregator_explicit):
+        assert aggregator_explicit.area_name_uuid_mapping == {}
+
+
+class TestStartWebsocketConnection:
+    """Includes methods which test start_websocket_connection method."""
+
+    @staticmethod
+    def test_start_websocket_connection__dispatcher(aggregator_explicit):
+        with patch("gsy_e_sdk.aggregator.AggregatorWebsocketMessageReceiver") \
+                as mocked_class:
+            aggregator_explicit.start_websocket_connection()
+            mocked_class.assert_called()
+
+    @staticmethod
+    def test_start_websocket_connection__websocket_creation(aggregator_explicit):
+        websocket_uri_test = f"{test_simulation['websockets_domain_name']}" \
+                             + f"/{test_simulation['uuid']}/aggregator/" \
+                             + f"{test_agg['uuid']}/"
+
+        with patch("gsy_e_sdk.aggregator.WebsocketThread") as mocked_class:
+            aggregator_explicit.aggregator_uuid = test_agg["uuid"]
+            aggregator_explicit.start_websocket_connection()
+            mocked_class.assert_called_with(websocket_uri_test, test_simulation["domain_name"],
+                                            aggregator_explicit.dispatcher)
+
+    @staticmethod
+    def test_start_websocket_connection__websocket_start(aggregator_explicit):
+        aggregator_explicit.start_websocket_connection()
+        aggregator_explicit.websocket_thread.start.assert_called_once()
+
+    @staticmethod
+    def test_start_websocket_connection__callback_thread(aggregator_explicit):
+        aggregator_explicit.start_websocket_connection()
+        assert aggregator_explicit.callback_thread == "test_set_callback_thread"
```

Comment:
Why not testing for the `ThreadPoolExecutor` like you did in the other tests:
```python
 with patch("gsy_e_sdk.aggregator.ThreadPoolExecutor") as mocked_class:
    aggregator_explicit.start_websocket_connection()
    mocked_class.assert_called_once_with(max_workers=MAX_WORKER_THREADS)
```
?